### PR TITLE
Track reference Sequence number for batched messages

### PR DESCRIPTION
**Pull request by vladsud**
Addressing https://dev.azure.com/fluidframework/internal/_queries/edit/1848/?triage=true (sort of)
Updated https://dev.azure.com/fluidframework/internal/_queries/edit/1834 as serious follow is required IMHO.
Thanks @agarwal-navin for pointing out, my original assumption was wrong, and assert (that is commented here) points to serious issues in how we process ops.

### CODE REVIEW

**Comment by agarwal-navin** [comment]
Could you please fix the typos in the title :)

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -2599,10 +2608,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             deserializedContent,
             metadata,
             localOpMetadata,
+            referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
         });
         if (this._flushMode !== FlushMode.TurnBased) {
             this.flush();
         } else if (!this.flushTrigger) {
+            this.flushTrigger = true;
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2616]: This looks like new code. Why is this needed?

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (vladsud) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2616]: I missed it in my previous check in. Without that we would have a lot of promises (one per on) that do empty flushes (as only one of them does work).
